inject/src/main/java/io/micronaut/context/env/PropertySourcePropertyResolver.java

```diff
@@ -835,29 +842,6 @@ public void close() throws Exception {
         }
     }
 
-
-    /**
-     * The property catalog to use.
-     */
-    protected enum PropertyCatalog {
```

Comment:
this is a breaking change for classes that subclass this class. Deprecate using `forRemoval` instead.

Reply:
Why it's breaking change? This enum was interanal, not public api. I just moved it to public API.

Reply:
it doesn't appear to be annotated with `@Internal`

Reply:
It is not annotated, but it has a scope of protected and cannot be used from outside.

Reply:
This code suggests otherwise:

```java
package example;

import io.micronaut.context.env.PropertySourcePropertyResolver;

public class TestResolver
    extends PropertySourcePropertyResolver {
    public TestResolver() {
        super();
        PropertyCatalog[] values = PropertyCatalog.values();
        System.out.println("values = " + values);
    }
}
```

Reply:
It turns out that I can't use PropertyCatalog in an annotation and I need to create a second PropertyCatalog enum that will be used specifically in the annotation. And do mapping between them. Right?

Reply:
or duplicate them. We follow semantic versioning which prohibits breaking changes in a minor release so we cannot merge changes that may break user code.

Reply:
Okay, I did everything, now the changes will be invisible

Reply:
![Uploading {D7CF0F89-FD03-45D4-BBE3-EE9B393BFEFF}.png…]()